File: frontend/src/api/screen.ts
import api from './client';
import type { ScreenResult, StrategyInfo } from './types';

export async function fetchStrategies(): Promise<StrategyInfo[]> {
  const { data } = await api.get<StrategyInfo[]>('/screen/strategies');
  return data;
}

export async function runScreen(strategy: string, limit = 20): Promise<ScreenResult> {
  const { data } = await api.post<ScreenResult>('/screen/run', { strategy, limit, use_parallel: true });
  return data;
}

File: api/routes/screen.py
"""选股筛选路由"""

from fastapi import APIRouter, HTTPException

from api.models.screen import ScreenRequest, ScreenResponse, StrategyInfo
from api.services import screen_service

router = APIRouter()


@router.get("/strategies", response_model=list[StrategyInfo])
def list_strategies():
    """列出所有可用选股策略"""
    return screen_service.get_strategies()


@router.post("/run", response_model=ScreenResponse)
def run_screen(req: ScreenRequest):
    """执行选股筛选"""
    try:
        return screen_service.run_screen(
            strategy=req.strategy,
            limit=req.limit,
            use_parallel=req.use_parallel,
        )
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"筛选失败: {e}")
